File: solutions/cpp/collatz-conjecture/1/collatz_conjecture.cpp
#include "collatz_conjecture.h"
#include <stdexcept>

namespace collatz_conjecture {

    int steps(int number)
    {
        int interations{};
        
        if (number < 1){throw std::domain_error("invalid number");}

        while (number > 1)
            {
                if (number % 2 == 0){number = number / 2;}
                else{number = (number * 3) + 1;}
                interations++;
            }
        return interations;
    }
// TODO: add your solution here

}  // namespace collatz_conjecture

File: solutions/ruby/collatz-conjecture/1/collatz_conjecture.rb
module CollatzConjecture
  def self.steps(n)
    raise ArgumentError if n <= 0
    
    i = 0
    
    while n != 1
      if n % 2 == 0
        n /= 2
      else
        n = n * 3 + 1
      end
      i += 1
    end
    
    i
  end
end